dashboard/app/helpers/users_helper.rb

```diff
@@ -86,13 +88,54 @@ def summarize_user_progress_for_all_scripts(user)
               result: result,
               submitted: submitted
           }
+
+          # Just in case this level has multiple pages, in which case we add an additional
+          # array of booleans indicating which pages have been completed.
+          pages_completed = get_pages_completed(user, sl)
+          user_data[:levels][sl.level_id][:pages_completed] = pages_completed if pages_completed
         end
       end
     end
 
     user_data
   end
 
+  # Given a user and a script-level, returns a nil if there is only one page, or an array of
+  # boolean values if there are multiple pages.  The array contains true for each page that
+  # is considered complete.  Since this is currently just used for multi-page LevelGroup levels,
+  # true means that a valid (though not necessarily correct) answer has been given for each
+  # level embedded on the page.
+  def get_pages_completed(user, sl)
+    level = sl.level
+
+    if level.type == "LevelGroup"
+      pages_completed = []
+
+      last_attempt = JSON.parse(user.last_attempt(level).try(:level_source).data)
+
+      # Go through each page.
+      level.properties["pages"].each do |page|
```

Comment:
Nit: [`serialized_attr` adds getter/setters](https://github.com/code-dot-org/code-dot-org/blob/8ee5d8fcf2a87dbc42004e23dee2d7af4c09567a/dashboard/app/models/concerns/serialized_properties.rb#L48-L52) so you can just do `level.pages ...` here (as long as `level.is_a? LevelGroup`).